### ISSUE TITLE
launchdarkly_feature_flag_environment apply does not remove rules

### ISSUE THREAD

**Issue by mlhoyt** [launchdarkly/terraform-provider-launchdarkly#45]
Using Terraform v0.12.29 the `plan`, after removing the one and only `rules` section from one particular flag environment configuration:
```
...
* provider.launchdarkly: version = "~> 1.5"
...

No changes. Infrastructure is up-to-date.
...
```

I verified the `rule` did exist on the LaunchDarkly side via the UI prior to the `plan` and `apply`.

The subsequent  `apply` did NOT make any changes and did NOT impact the `rule` that existed in the UI.

A subsequent `plan` did not want to make any changes.

I manually removed the `rule` via the UI and ran another `plan` which still did not want to make any changes.

Things behaved here like the Terraform state had no idea about the existence of the flag environment rule which was created previously (earlier today) via Terraform.

**Comment by ldhenry** [comment]
Hi @mlhoyt,

Thanks for bringing this to our attention. It definitely sounds like a provider bug. I've filed an internal ticket to get this bug fixed and put into our next release. I'll be sure to update this issue when when the release is ready.

Thanks,
Henry

**Comment by sloloris** [comment]
Hi @mlhoyt, I've dug into our code a bit and this is actually the intended behavior for deleting optional fields from the config. We will try to address this better in our documentation early in the new year but, in the meantime, could you try updating with an empty rules block `rules {}` instead of deleting the rules block altogether and let me know if that works for you?

**Comment by mlhoyt** [comment]
Hi @sloloris,

Thank you for the response and insight.  I don't know that an empty rules block would have occurred to me so documentation around that would definitely be helpful.

I don't have a scenario to test this with right now but I will remember this when the situation presents itself again.

Does it make sense to leave this issue open until the documentation is updated?

**Comment by mlhoyt** [comment]
Hi @sloloris,

We finally had a scenario to try the empty rules block solution (to remove a previous rule definition) and it did NOT work:

Terraform version: `0.13.6`
LaunchDarkly provider version: `v1.5.0`
```
Error: failed to update flag "<FLAG-KEY>" in project "<PROJECT>", environment "<ENVIRONMENT>": 400 Bad Request: {"code":"invalid_request","message":"rule 1/1: rule must have at least 1 clause"}
```

It appears that a `rules` block requires at least one `clause` block and each `clause` block requires several fields.

For now we have used the previous solution of removing the `rules {}` from the Terraform and manually removing the rule via the UI.

**Comment by mlhoyt** [comment]
Hi @sloloris and @ldhenry,

Curious if you have any feedback on the issue we encountered when trying the "empty rules block" solution (to remove a previous rule definition)?

**Comment by mlhoyt** [comment]
Hi @sloloris and @ldhenry ,

Checking-in to see if you have any feedback on the issue we encountered when trying the "empty rules block" solution (to remove a previous rule definition)?

**Comment by sloloris** [comment]
Hi @mlhoyt,

First off, many apologies for the delay. We've investigated a fix for this but believe it may introduce a breaking change that we would have to roll into the v2 release. However, we do have one other solution that we are investigating that may be compatible with v1 - we can't promise that it will work, but we are testing it out now so we should be able to get you an update next week.

Have a great weekend!

**Comment by ldhenry** [comment]
Hi @mlhoyt,

We spent some more time investigating this issue and unfortunately we were not able to find a solution that didn't introduce a breaking change. This is on the roadmap for our v2 release, and we'll update this issue when it is available.

Thanks,
Henry

**Comment by sloloris** [comment]
Hi @mlhoyt! 

We're very pleased to let you know that we've finally released v2.0.0 of our provider with a fix for this very issue! In addition, we've streamlined our resources so that this convention will generally apply (removal will result in removal from the LaunchDarkly resource and should be reflected in the UI). We will be releasing an upgrade guide soon on our [docs website](https://docs.launchdarkly.com/integrations/terraform). Please let us know if you have any questions.
